lib/internal/mime.js

```diff
@@ -0,0 +1,369 @@
+      const rawValue = valueEnd === -1 ?
+        StringPrototypeSlice(str, position) :
+        StringPrototypeSlice(str, position, valueEnd);
+      position += rawValue.length;
+      const trimmedValue = StringPrototypeSlice(
+        rawValue,
+        0,
+        SafeStringPrototypeSearch(rawValue, START_ENDING_WHITESPACE)
+      );
+      // Ignore parameters without values
+      if (trimmedValue === '') continue;
+      parameterValue = trimmedValue;
+    }
+    if (
+      parameterString !== '' &&
+      SafeStringPrototypeSearch(parameterString,
+                                NotHTTPTokenCodePoint) === -1 &&
+      SafeStringPrototypeSearch(parameterValue,
+                                NotHTTPQuotedStringCodePoint) === -1 &&
+      paramsMap.has(parameterString) === false
+    ) {
+      paramsMap.set(parameterString, parameterValue);
+    }
+    position++;
+  }
+  return paramsMap;
+}
+
+function escapeQuoteOrSolidus(str) {
+  let result = '';
+  for (let i = 0; i < str.length; i++) {
+    const char = str[i];
+    result += (char === '"' || char === '\\') ? `\\${char}` : char;
+  }
+  return result;
+}
+
+const encode = (value) => {
+  if (value.length === 0) return '""';
+  const encode = SafeStringPrototypeSearch(value, NotHTTPTokenCodePoint) !== -1;
+  if (!encode) return value;
+  const escaped = escapeQuoteOrSolidus(value);
+  return `"${escaped}"`;
+};
+
+class MIMEParams {
+  #data = new SafeMap();
+
+  delete(name) {
+    this.#data.delete(name);
+  }
+
+  get(name) {
+    const data = this.#data;
+    if (data.has(name)) {
+      return data.get(name);
+    }
+    return null;
+  }
+
+  has(name) {
+    return this.#data.has(name);
+  }
+
+  set(name, value) {
+    const data = this.#data;
+    name = `${name}`;
+    value = `${value}`;
+    const invalidNameIndex = SafeStringPrototypeSearch(name, NotHTTPTokenCodePoint);
+    if (name.length === 0 || invalidNameIndex !== -1) {
+      throw new ERR_INVALID_MIME_SYNTAX(
+        'parameter name',
+        name,
+        invalidNameIndex,
+      );
+    }
+    const invalidValueIndex = SafeStringPrototypeSearch(
+      value,
+      NotHTTPQuotedStringCodePoint);
+    if (invalidValueIndex !== -1) {
+      throw new ERR_INVALID_MIME_SYNTAX(
+        'parameter value',
+        value,
+        invalidValueIndex,
+      );
+    }
+    data.set(name, value);
+  }
+
+  *entries() {
+    yield* this.#data.entries();
+  }
+
+  *keys() {
+    yield* this.#data.keys();
+  }
+
+  *values() {
+    yield* this.#data.values();
+  }
+
+  toString() {
+    let ret = '';
+    for (const { 0: key, 1: value } of this.#data) {
+      const encoded = encode(value);
+      // Ensure they are separated
+      if (ret.length) ret += ';';
+      ret += `${key}=${encoded}`;
+    }
+    return ret;
+  }
+
+  // Used to act as a friendly class to stringifying stuff
+  // not meant to be exposed to users, could inject invalid values
+  static _data(o) {
```

Comment:
If this is meant to be private, should it use the private method syntax? `#data`?

Reply:
See https://github.com/nodejs/node/pull/21128#discussion_r996353138, it's being deleted to avoid being exposed, but we actually need it to be on defined in the class body to access the private property.

Reply:
Okay, then maybe let’s just add a comment explaining that? I feel like someone might try to refactor it otherwise.

Reply:
So I've removed the underscore trailing method as it was only used by one function, let me know if you think it needs more.